helix-view/src/register.rs

```diff
@@ -0,0 +1,355 @@
+use std::{borrow::Cow, collections::HashMap};
+
+use anyhow::Result;
+use helix_core::hashmap;
+
+use crate::{clipboard::ClipboardType, document::SCRATCH_BUFFER_NAME, Editor};
+
+pub const SPECIAL_REGISTERS: [char; 6] = ['_', '#', '.', '%', '*', '+'];
+
+pub trait Register: std::fmt::Debug {
+    fn name(&self) -> char;
+    fn preview(&self) -> &str;
+
+    fn read(&self, editor: &Editor) -> Vec<String>;
+
+    fn write(&mut self, _editor: &mut Editor, _values: Vec<String>) -> Result<()> {
+        Ok(())
+    }
```

Comment:
you provide a default implementation that returns `Ok(())` and therefore essentially acts like a blackhole register. I think a better default would be to return an error and then overwrite the default implementation for the blackhole register (so that it acts like a noop). Its not super critical but I think register that can't be written to should return an error instead of acting like a blackhole register.

Reply:
This was just being lazy and not wanting to duplicate these 6 LoC for the read-only registers 😅. It's a weird default though so I'll just use these lines in the read-only registers

Reply:
Being explicit is also good :+1: I would remove the default implementation entirely, in that case, though to avoid accidentally forgetting to implement the method